### PR TITLE
BUGFIX: TMA-1405 Mask hidden parameters only in K8s execution

### PR DESCRIPTION
**Pull request by sadam21**

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1371/ : SUCCESS in 10s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/95/ : SUCCESS in 14m 10s

File: lib/gooddata/bricks/middleware/logger_middleware.rb [hunk 1]
```diff
@@ -38,7 +38,8 @@ def call(params)
           FileUtils.mkpath log_directory
           logger = Logger.new("#{log_directory}/#{execution_id}.log")
           logger.level = params['GDC_LOG_LEVEL'] || 'info'
-          logger = MaskLoggerDecorator.new(logger, params)
+          values_to_mask = params['values_to_mask'] || []
+          logger = MaskLoggerDecorator.new(logger, values_to_mask) if values_to_mask.any?
```
Comment (kubamahnert) [inline lib/gooddata/bricks/middleware/logger_middleware.rb]: let's keep the interface of GoodData.logger constant and always initialize MaskLoggerDecorator, even if values_to_mask is an empty array.

Comment (sadam21) [reply lib/gooddata/bricks/middleware/logger_middleware.rb]: Done

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1372/ : SUCCESS in 25s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1373/ : SUCCESS in 33s

**Comment by sadam21** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1375/ : SUCCESS in 19s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/286/ : SUCCESS in 25s

**Pull request merged**